app/models/user.rb

```diff
@@ -40,8 +40,18 @@ class User
   validates_processing_of :avatar
 
   has_many :histories, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
 
   def will_save_change_to_email?
     false
   end
+
+  def liked?(book_id)
+    likes.where(book_id: book_id).first.present?
+  end
+
+  def user_like(book_id)
+    likes.where(book_id: book_id).first
+  end
```

Comment:
You got 2 almost the same methods.
```ruby
likes.where(book_id: book_id)
```
reuse one `user_like` merhod
```ruby
def liked?(book_id)
  user_like(book_id).present?
end
def user_like(book_id)
  likes.where(book_id: book_id).first
end
```

Reply:
fixed